--- a/lib/servy/handler.ex
+++ b/lib/servy/handler.ex
@@ -1,11 +1,14 @@
 defmodule Servy.Handler do
   def handle(request) do
     request 
-    |> parse 
+    |> parse
+    |> log 
     |> route 
     |> format_response
   end
 
+  def log(conv), do: IO.inspect conv
+
   def parse(request) do
     [method, path, _] =
       request 
@@ -17,7 +20,15 @@ defmodule Servy.Handler do
   end
 
   def route(conv) do
-    %{ conv | resp_body: "Bears, Lions, Tigers" }
+    route(conv, conv.method, conv.path)
+  end
+
+  def route(conv, "GET", "/wildthings") do
+    %{ conv | resp_body: "Bears, Lions, Tigers" }          
+  end
+
+  def route(conv, "GET", "/bears") do
+    %{ conv | resp_body: "Teddy, Smokey, Paddington" }    
   end
 
   def format_response(conv) do
@@ -44,5 +55,28 @@ response = Servy.Handler.handle(request)
 
 IO.puts response
 
+request = """
+GET /bears HTTP/1.1
+Host: example.com
+User-Agent: ExampleBrowser/1.0
+Accept: */*
+
+"""
+
+response = Servy.Handler.handle(request)
+
+IO.puts response
+
+request = """
+GET /bigfoot HTTP/1.1
+Host: example.com
+User-Agent: ExampleBrowser/1.0
+Accept: */*
+
+"""
+
+response = Servy.Handler.handle(request)
+
+IO.puts response
 
 
