--- a/lib/servy/handler.ex
+++ b/lib/servy/handler.ex
@@ -4,6 +4,9 @@ defmodule Servy.Handler do
 
   @pages_path Path.expand("../../pages", __DIR__)
 
+  import Servy.Plugins, only: [rewrite_path: 1, log: 1, track: 1]
+  import Servy.Parser, only: [parse: 1]
+
   @doc "Transforms the request into a response."
   def handle(request) do
     request 
@@ -15,36 +18,6 @@ defmodule Servy.Handler do
     |> format_response
   end
 
-  @doc "Logs 404 requests"
-  def track(%{status: 404, path: path} = conv) do
-    IO.puts "Warning: #{path} is on the loose!"
-    conv
-  end
-
-  def track(conv), do: conv
-
-  def rewrite_path(%{path: "/wildlife"} = conv) do
-    %{ conv | path: "/wildthings" }
-  end
-
-  def rewrite_path(conv), do: conv
-
-  def log(conv), do: IO.inspect conv
-
-  def parse(request) do
-    [method, path, _] =
-      request 
-      |> String.split("\n") 
-      |> List.first    
-      |> String.split(" ")
-
-    %{ method: method, 
-       path: path, 
-       resp_body: "",
-       status: nil
-     }
-  end
-
   def route(%{ method: "GET", path: "/wildthings" } = conv) do
     %{ conv | status: 200, resp_body: "Bears, Lions, Tigers" }          
   end
new file mode 100644
--- /dev/null
+++ b/lib/servy/parser.ex
@@ -0,0 +1,15 @@
+defmodule Servy.Parser do
+  def parse(request) do
+    [method, path, _] =
+      request 
+      |> String.split("\n") 
+      |> List.first    
+      |> String.split(" ")
+
+    %{ method: method, 
+       path: path, 
+       resp_body: "",
+       status: nil
+     }
+  end
+end
new file mode 100644
--- /dev/null
+++ b/lib/servy/plugins.ex
@@ -0,0 +1,17 @@
+defmodule Servy.Plugins do
+  @doc "Logs 404 requests"
+  def track(%{status: 404, path: path} = conv) do
+    IO.puts "Warning: #{path} is on the loose!"
+    conv
+  end
+
+  def track(conv), do: conv
+
+  def rewrite_path(%{path: "/wildlife"} = conv) do
+    %{ conv | path: "/wildthings" }
+  end
+
+  def rewrite_path(conv), do: conv
+
+  def log(conv), do: IO.inspect conv
+end
