--- a/lib/servy/handler.ex
+++ b/lib/servy/handler.ex
@@ -7,8 +7,13 @@ defmodule Servy.Handler do
   end
 
   def parse(request) do
-    # TODO: Parse the request string into a map:
-    conv = %{ method: "GET", path: "/wildthings", resp_body: "" }
+    [method, path, _] =
+      request 
+      |> String.split("\n") 
+      |> List.first    
+      |> String.split(" ")
+
+    %{ method: method, path: path, resp_body: "" }
   end
 
   def route(conv) do
